### DISCUSSION TITLE
Support GitHub Light/Dark Image Switcher

### DISCUSSION THREAD

**Discussion by Sherlouk** [SwiftPackageIndex/SwiftPackageIndex-Server#1397] [Ideas]
GitHub have recently announced a new feature which allows for markdown documents (such as the READMEs we render) to support images which change based on light/dark mode.

https://twitter.com/ashtom/status/1463554157932023808

```
![GitHub-Mark-Light](https://user-images.githubusercontent.com/3369400/139447912-e0f43f33-6d9f-45f8-be46-2df5bbc91289.png#gh-dark-mode-only)![GitHub-Mark-Dark](https://user-images.githubusercontent.com/3369400/139448065-39a229ba-4b06-434b-bc67-616e2ed80c8f.png#gh-light-mode-only)
```

https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#specifying-the-theme-an-image-is-shown-to

I don't know how much work it will be to handle this, or indeed what format we get it back in the HTML from GitHub (it might even support it for free!) but would be good to look into this.

**Comment by daveverwer** [comment]
Yep. I'd be very much in favour of supporting this if we can make it work without too much trouble.

Based on doing no research at all, 😱 my gut feeling, given the way that we grab rendered README files from GitHub that all of the requisite parts of this might already be coming in and we might just need a CSS tweak to support this. However, It'd be great to see an example of a repository that's using it to know. Do we have any examples of README files in the index that use this? If not, can we create one using one of our packages? [LeftPad](https://swiftpackageindex.com/daveverwer/LeftPad) is available for testing at a push.

**Reply by Sherlouk** [reply]
I'd agree with your research-less assessment, no idea of any repos using it - it's only just been announced.

Let's update LeftPad for now.

**Reply by daveverwer** [reply]
> Let's update LeftPad for now.

Are there no images in any of your real repos?

**Reply by Sherlouk** [reply]
Not that have light/dark versions. Suppose I could make some graphics for one of them

**Reply by MortenGregersen** [reply]
I have a package where the logo in the README is now using this functionality: https://github.com/FluxorOrg/Fluxor
It is also on Swift Package Index: https://swiftpackageindex.com/FluxorOrg/Fluxor

**Reply by daveverwer** [reply]
… and there we have an answer!

<img width="873" alt="Screenshot 2021-11-25 at 20 36 09@2x" src="https://user-images.githubusercontent.com/5180/143497886-17b693f5-81ba-43e5-ba1e-233aa1549a4f.png">

I'll convert this into an issue and get it fixed.

**Reply by MortenGregersen** [reply]
Maybe it's not an issue if you add the image via markdown. This logo is added with HTML in the markdown so it can be centered. Maybe that could play a part in how Github serves the HTML.

**Reply by daveverwer** [reply]
I wish they had done it by using a `picture` element, like this:

https://github.com/SwiftPackageIndex/SwiftPackageIndex-Blog/blob/master/Content/posts/launching-the-swift-package-index-playgrounds-app-for-macos.md?plain=1#L17

I think it's going to need JavaScript on the SPI site the way they've done it.

**Reply by daveverwer** [reply]
It's possible CSS can target the anchor at the end of a URL, but I'm not sure.

**Reply by Sherlouk** [reply]
Can we inspect element in GitHub.com to see how they're doing it?

Is their DOM different to what we get? What CSS do they have?

**Reply by daveverwer** [reply]
Yep, of course. I haven't started looking at a solution yet.

**Comment by daveverwer** [comment]
Filed as #1398
